packages/@o3r/core/schematics/ng-add/presets/helpers.ts

```diff
@@ -0,0 +1,77 @@
+import { chain, externalSchematic, Rule, Tree } from '@angular-devkit/schematics';
+import type { JsonValue } from '@angular-devkit/core';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { NodePackageName } from '@angular-devkit/schematics/tasks/package-manager/options';
+import { lastValueFrom } from 'rxjs';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import type { PackageJson } from 'type-fest';
+import type { PresetOptions } from './preset.interface';
+
+/**
+ * Install dev dependency on your application
+ *
+ * Note: it should not be moved to other packages as it should run before the installation
+ * of peer dependencies
+ */
+export class DevInstall extends NodePackageInstallTask {
+  public quiet = false;
+
+  /** @inheritdoc */
+  public toConfiguration() {
+    const installOptions = process.env?.npm_execpath?.includes('yarn') ? 'yarn' : 'npm';
+    return {
+      name: NodePackageName,
+      options: {
+        command: 'install',
+        quiet: this.quiet,
+        workingDirectory: this.workingDirectory,
+        packageName: `${this.packageName!} ${installOptions === 'yarn' ? '--prefer-dev' : '-D'}`,
+        packageManager: installOptions
+      }
+    };
+  }
+}
+
+/**
+ * Set options to a schematics label
+ *
+ * @param tree Angular Schematic Tree
+ * @param params Options to set as global
+ * @param schematicLabel Label of the schematics options to set
+ */
+export function addSchematicsOptions(tree: Tree, params: Record<string, JsonValue | undefined>, schematicLabel = '*:ng-add'): Tree {
```

Comment:
couldn't you use this https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/schematics/src/rule-factories/ng-add/index.ts#L154 instead ?